src/compiler/checker.ts

```diff
@@ -6423,14 +6427,14 @@ namespace ts {
                 const classType = declaration.kind === SyntaxKind.Constructor ?
                     getDeclaredTypeOfClassOrInterface(getMergedSymbol((<ClassDeclaration>declaration.parent).symbol))
                     : undefined;
-                const typeParameters = classType ? classType.localTypeParameters : getTypeParametersFromDeclaration(declaration);
+                const typeParameters = classType ? classType.localTypeParameters : declaration.kind === SyntaxKind.JSDocFunctionType ? emptyArray : getTypeParametersFromDeclaration(declaration);

```

Comment:
seems like `getTypeParametersFromDeclaration` could handle jsdoc functions too, since it calls `getEffectiveTypeParameters` to provide precisely that level of abstraction.